fuse_core/include/fuse_core/graph_deserializer.hpp

```diff
@@ -74,7 +74,7 @@ class GraphDeserializer
    * @return          A unique_ptr to a derived Graph object
    */
   inline fuse_core::Graph::UniquePtr deserialize(
-    const fuse_msgs::msg::SerializedGraph::ConstSharedPtr msg) const
+    const fuse_msgs::msg::SerializedGraph::SharedPtr msg) const
```

Comment:
Since this is just passing through to a method that treats the value as const, I think `ConstSharedPtr` makes more sense as it would allow both `std::shared_ptr<const MessageType>` and `std::shared_ptr<MessageType>` to be pasesd in.

<details><summary> Small example in C++</summary>

```c++
#include <iostream>
#include <memory>

void const_func(std::shared_ptr<const double> dp)
{
  std::cout << "Const Got some double: " << *dp << "\n";
}

void non_const_func(std::shared_ptr<double> dp)
{
  std::cout << "non Const Got some double: " << *dp << "\n";
}


int main()
{
  auto const_dp = std::make_shared<const double>(42.0);
  auto non_const_dp = std::make_shared<double>(13.0);

  const_func(const_dp);
  const_func(non_const_dp);


  // Can't do this
  // non_const_func(const_dp);
  //
  // because:
  //    $ g++ -std=c++17 a.cpp 
  //    a.cpp: In function ‘int main()’:
  //    a.cpp:23:18: error: could not convert ‘const_dp’ from ‘shared_ptr<const double>’ to ‘shared_ptr<double>’
  //       23 |   non_const_func(const_dp);
  //          |                  ^~~~~~~~
  //          |                  |
  //          |                  shared_ptr<const double>
  //

  non_const_func(non_const_dp);

  return 0;
}
```
</details>